code/modules/client/preference_setup/loadout/loadout.dm

```diff
@@ -41,27 +41,57 @@ var/list/gear_datums = list()
 /datum/category_item/player_setup_item/loadout/load_character(var/savefile/S)
 	S["gear"] >> pref.gear
 	S["gear_list"] >> pref.gear_list
-	if(pref.gear_list!=null && pref.gear_slot!=null)
-		pref.gear = pref.gear_list["[pref.gear_slot]"]
-	else
-		S["gear"] >> pref.gear
+
+/datum/category_item/player_setup_item/loadout/load_character_special(var/savefile/S)
+	pref.gear_modified = FALSE
+	//Query the ss13_characters_gear table
+	if(GLOB.config.sql_saves && establish_db_connection(GLOB.dbcon))
+		var/fetched_gear = list()
+
+		var/DBQuery/character_gear_query = GLOB.dbcon.NewQuery("SELECT slot, name, tweaks FROM ss13_characters_gear WHERE char_id = :char_id:")
+		character_gear_query.Execute(list("char_id"=pref.current_character))
+		while(character_gear_query.NextRow())
+			var/slot = character_gear_query.item[1]
+			var/name = character_gear_query.item[2]
+
+			if(!fetched_gear[slot])
+				fetched_gear[slot] = list()
+
+			var/tweaks = list()
+			if(character_gear_query.item[3])
+				tweaks = json_decode(character_gear_query.item[3])
+
+			fetched_gear[slot][name]=tweaks
+
+		pref.gear_list = fetched_gear
+	return
+
+/datum/category_item/player_setup_item/loadout/save_character_special(var/savefile/S)
+	//Persist the loadout to the database
+	if(GLOB.config.sql_saves && establish_db_connection(GLOB.dbcon) && pref.gear_modified)
```

Comment:
this check is already duplicated right after it's called in `save_character`, I'd probably just not have the check here and only run it there? Or collapse all the checks into the procs

Reply:
save_character_special mirrors the usecase of load_character_special but for saving instead of loading.

save/load_charracter_special is supposed to be always called for every player setup item regardless if sql or file saves are used.